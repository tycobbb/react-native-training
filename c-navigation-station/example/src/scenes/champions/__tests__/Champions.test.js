/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { Champions } from '../Champions'
import { fetchChampions } from '../fetchChampions'
import { championFixtures } from 'fixtures'

// mocks
jest.mock('../fetchChampions')

// setup
let subject

function loadSubject (props, options) {
  subject = shallow(<Champions {...props} />, options)
}

// spec
afterEach(() => {
  subject = null
})

describe('#render', () => {
  it('renders properly when loading', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })

  it('renders properly when loading is finished', () => {
    loadSubject()
    subject.setState({ isLoading: false, champions: championFixtures })
    expect(subject).toMatchSnapshot()
  })
})

describe('#renderChampion', () => {
  it('renders properly', () => {
    loadSubject()
    const champion = championFixtures[0]
    const view = shallow(subject.instance().renderChampion({ item: champion }))
    expect(view).toMatchSnapshot()
  })
})

describe('#componentDidMount', () => {
  it('finishes loading after fetching champions', async () => {
    const champions = ['champ-1', 'champ-2']
    fetchChampions.mockReturnValueOnce(Promise.resolve(champions))
    loadSubject({}, { disableLifecycleMethods: true })

    await subject.instance().componentDidMount()
    expect(subject.state('isLoading')).toBeFalsy()
    expect(subject.state('champions')).toEqual(champions)
  })
})

describe('when the champion is tapped', () => {
  it('shows the champion screen', () => {
    const navigate = jest.fn()
    loadSubject({ navigation: { navigate } })

    const champion = championFixtures[0]
    subject.instance().didTapChampion(champion)

    expect(navigate).toHaveBeenCalledWith('champion', { id: champion.id })
  })
})
