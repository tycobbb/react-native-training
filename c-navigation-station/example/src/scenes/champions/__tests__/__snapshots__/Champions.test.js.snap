// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders properly when loading 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Champions />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="large"
        />
      </View>,
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="large"
        />,
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "animating": true,
          "color": "#999999",
          "hidesWhenStopped": true,
          "size": "large",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="large"
          />
        </View>,
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="large"
          />,
          "style": Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "animating": true,
            "color": "#999999",
            "hidesWhenStopped": true,
            "size": "large",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`#render renders properly when loading is finished 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Champions />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <FadeIn>
        <FlatList
          data={
            Array [
              Object {
                "id": 1,
                "image": Object {
                  "full": "Annie.png",
                  "group": "champion",
                  "h": 48,
                  "sprite": "champion0.png",
                  "w": 48,
                  "x": 288,
                  "y": 0,
                },
                "key": "Annie",
                "name": "Annie",
                "title": "the Dark Child",
              },
            ]
          }
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          style={undefined}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />
      </FadeIn>,
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": <FlatList
          data={
            Array [
              Object {
                "id": 1,
                "image": Object {
                  "full": "Annie.png",
                  "group": "champion",
                  "h": 48,
                  "sprite": "champion0.png",
                  "w": 48,
                  "x": 288,
                  "y": 0,
                },
                "key": "Annie",
                "name": "Annie",
                "title": "the Dark Child",
              },
            ]
          }
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          style={undefined}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "data": Array [
            Object {
              "id": 1,
              "image": Object {
                "full": "Annie.png",
                "group": "champion",
                "h": 48,
                "sprite": "champion0.png",
                "w": 48,
                "x": 288,
                "y": 0,
              },
              "key": "Annie",
              "name": "Annie",
              "title": "the Dark Child",
            },
          ],
          "disableVirtualization": false,
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "style": undefined,
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <FadeIn>
          <FlatList
            data={
              Array [
                Object {
                  "id": 1,
                  "image": Object {
                    "full": "Annie.png",
                    "group": "champion",
                    "h": 48,
                    "sprite": "champion0.png",
                    "w": 48,
                    "x": 288,
                    "y": 0,
                  },
                  "key": "Annie",
                  "name": "Annie",
                  "title": "the Dark Child",
                },
              ]
            }
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            style={undefined}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
        </FadeIn>,
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <FlatList
            data={
              Array [
                Object {
                  "id": 1,
                  "image": Object {
                    "full": "Annie.png",
                    "group": "champion",
                    "h": 48,
                    "sprite": "champion0.png",
                    "w": 48,
                    "x": 288,
                    "y": 0,
                  },
                  "key": "Annie",
                  "name": "Annie",
                  "title": "the Dark Child",
                },
              ]
            }
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            style={undefined}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "data": Array [
              Object {
                "id": 1,
                "image": Object {
                  "full": "Annie.png",
                  "group": "champion",
                  "h": 48,
                  "sprite": "champion0.png",
                  "w": 48,
                  "x": 288,
                  "y": 0,
                },
                "key": "Annie",
                "name": "Annie",
                "title": "the Dark Child",
              },
            ],
            "disableVirtualization": false,
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "style": undefined,
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`#renderChampion renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Champion
    champion={
      Object {
        "id": 1,
        "image": Object {
          "full": "Annie.png",
          "group": "champion",
          "h": 48,
          "sprite": "champion0.png",
          "w": 48,
          "x": 288,
          "y": 0,
        },
        "key": "Annie",
        "name": "Annie",
        "title": "the Dark Child",
      }
    }
    onPress={[Function]}
    style={
      Object {
        "marginBottom": 5,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": Array [
        <FadeInImage
          source={
            Object {
              "uri": "https://ddragon.leagueoflegends.com/cdn//7.10.1/img/champion/Annie.png",
            }
          }
          style={
            Object {
              "height": 80,
              "marginRight": 10,
              "width": 80,
            }
          }
        />,
        <SlideIn>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "300",
                "marginBottom": 5,
              }
            }
          >
            Annie
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "fontWeight": "600",
              }
            }
          >
            the Dark Child
          </Text>
        </SlideIn>,
      ],
      "onPress": [Function],
      "style": Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 80,
        },
        Object {
          "marginBottom": 5,
        },
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "source": Object {
            "uri": "https://ddragon.leagueoflegends.com/cdn//7.10.1/img/champion/Annie.png",
          },
          "style": Object {
            "height": 80,
            "marginRight": 10,
            "width": 80,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "300",
                  "marginBottom": 5,
                }
              }
            >
              Annie
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "fontWeight": "600",
                }
              }
            >
              the Dark Child
            </Text>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Annie",
              "disabled": false,
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 16,
                "fontWeight": "300",
                "marginBottom": 5,
              },
            },
            "ref": null,
            "rendered": "Annie",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "the Dark Child",
              "disabled": false,
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 12,
                "fontWeight": "600",
              },
            },
            "ref": null,
            "rendered": "the Dark Child",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": Array [
          <FadeInImage
            source={
              Object {
                "uri": "https://ddragon.leagueoflegends.com/cdn//7.10.1/img/champion/Annie.png",
              }
            }
            style={
              Object {
                "height": 80,
                "marginRight": 10,
                "width": 80,
              }
            }
          />,
          <SlideIn>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "300",
                  "marginBottom": 5,
                }
              }
            >
              Annie
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "fontWeight": "600",
                }
              }
            >
              the Dark Child
            </Text>
          </SlideIn>,
        ],
        "onPress": [Function],
        "style": Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 80,
          },
          Object {
            "marginBottom": 5,
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "source": Object {
              "uri": "https://ddragon.leagueoflegends.com/cdn//7.10.1/img/champion/Annie.png",
            },
            "style": Object {
              "height": 80,
              "marginRight": 10,
              "width": 80,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "300",
                    "marginBottom": 5,
                  }
                }
              >
                Annie
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 12,
                    "fontWeight": "600",
                  }
                }
              >
                the Dark Child
              </Text>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Annie",
                "disabled": false,
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 16,
                  "fontWeight": "300",
                  "marginBottom": 5,
                },
              },
              "ref": null,
              "rendered": "Annie",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "the Dark Child",
                "disabled": false,
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 12,
                  "fontWeight": "600",
                },
              },
              "ref": null,
              "rendered": "the Dark Child",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
